action.yml

```diff
@@ -0,0 +1,51 @@
+name: 'Leviathan Action'
+description: 'Test your software directly on hardware using the Levaithan Testing framework'
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: (Virtual) Leviathan build & test 
+      working-directory: ${{ env.LEVIATHAN_ROOT }}/
+      if: env.WORKER_TYPE == 'qemu'
+      shell: bash
+      run: |
+        make config
+        make build
+        make local-test QEMU_SECUREBOOT=${QEMU_SECUREBOOT} FLASHER_SECUREBOOT=${FLASHER_SECUREBOOT} || exit 1
+
+    - name: (Autokit) Leviathan build & test
+      working-directory: ${{ env.LEVIATHAN_ROOT }}/
+      if: env.WORKER_TYPE == 'testbot' || env.WORKER_TYPE == 'autokit'
+      shell: bash
+      run: |
+        make config
+        make build
```

Comment:
We could move the common steps, `make config` and `make build` to their own job that runs for both worker types IMO. Will also separate the prep logs from the test logs.

Reply:
I really would like that. That's actually much better for the UX. Should we do, 

1. make config, make build - one composite action for the build?
2. make test - the other compose site action

What do you mean by runs for both workers?

Reply:
sorry, I meant make them a step (not job) that runs regardless of worker type, before the `make test` that is different for each worker type

Reply:
What difference is it gonna make apart from getting them to be common which is a good suggest. I would rather have a UX improvement so that no one has to scroll pages worth of logs from Leviathan build. Any suggestions on how to accomplish that if at all possible?

Reply:
You're right, in a composite action all the logs are smashed together anyway. I was thinking of a job where each step is separate.

What you have will do